README.md

```diff
@@ -1,39 +1,120 @@
 # PSI Cardinality
 Private Set Intersection Cardinality protocol based on ECDH and Bloom Filters.
 
+## Requirements
+
+There are requirements for the entire project which each language shares. There also could be requirements for each target language:
+
+### Global Requirements
+These are the common requirements across all target languages of this project.
+
+- [Bazel](https://bazel.build)
+
+
+### Target Requirements
+Some target languages have additional requirements
+
+#### C++
+
+- None
+
+#### JavaScript
+
+- [NodeJS](https://nodejs.org/en/)
+- [Yarn](https://yarnpkg.com/)
+
+
 ## Compiling and Running
-The only requirement is [Bazel](https://bazel.build). To compile, run
+
+The repository uses a folder structure to isolate the supported targets from one another:
+
 ```
-bazel build //...
+src/<target language>/<sources>
 ```
-To run tests, use
+
+### C++
+
+Build all libraries (with all optimization levels), with a specific optimization level (-c opt), or a specific module
+
 ```
-bazel test //...
+# Build everything wth all optimization levels
```

Comment:
Omitting the `-c` flag doesn't build for all optimization levels, it just uses the [`fastbuild`](https://docs.bazel.build/versions/master/user-manual.html#flag--compilation_mode) config.